lib/sqlalchemy/sql/compiler.py

```diff
@@ -5166,8 +5179,29 @@ def get_column_specification(self, column, **kwargs):
             not column.identity or not self.dialect.supports_identity_columns
         ):
             colspec += " NOT NULL"
+
+        if column.system_versioning:
+            option = self.get_column_versioning_options(self, column)
+            colspec += " %s" % option
+
         return colspec
 
+    def get_column_versioning_options(self, *args, **kw) -> str:
+        """Return options for system versioning
+        Allows enabling or disabling, or configuring "GENERATED" statements"""
+        column = args[1]
+        if column.system_versioning == False:
```

Comment:
column.system_versioning should be an underscored attribute, however, there should be one system of detecting how we are rendering here, like if it's table._versioning_columns or an attribute on the column itself.  I sort of prefer table._versioning_columns, which would be populated within Period._set_parent().   overall there should be just one place we can look to see what we are rendering for each column, so perhaps table._versioning_columns could be some kind of structured data that records everything we need to know in order to render a column (we would use tests like "`versioning = table._versioning_info.get(column, None);   if versioning is not None and versioning.GENERATE_ROW_START:  return "GENERATED ALWAYS AS ROW START"`, something like that .

Reply:
I think this is a lot better now, I wound up getting just turning Table.periods into a collection and adding the _system_versioning_period helper property to return the `Period` with SV, rather than updating _versioning_columns